framework/source/class/qx/ui/form/VirtualSelectBox.js

```diff
@@ -521,53 +529,75 @@ qx.Class.define("qx.ui.form.VirtualSelectBox",
     {
       var highlightAppearance =
         qx.theme.manager.Appearance.getInstance().styleFrom("list-search-highlight");
+      // default style        
+      if (!highlightAppearance) {
+        this.debug('The current theme is missing the "list-search-highlight" appearance setting, using default.');
+        highlightAppearance = {
+          backgroundColor: 'rgba(255, 251, 0, 0.53)',
+          textDecorationStyle: 'dotted',
+          textDecorationLine: 'underline'
+        };
+      }
       var highlightStyle = '', styles = [];
-      if (highlightAppearance != null) {
-        var keys = Object.keys(highlightAppearance);
-        for (var k=0; k< keys.length; k++) {
-          var key = qx.module.util.String.hyphenate(keys[k]);
-          styles.push(key + ':' + highlightAppearance[keys[k]]);
-        }
-        highlightStyle = styles.join(';');
+      var keys = Object.keys(highlightAppearance);
+      for (var k=0; k< keys.length; k++) {
+        var key = qx.module.util.String.hyphenate(keys[k]);
+        styles.push(key + ':' + highlightAppearance[keys[k]]);
       }
+      highlightStyle = styles.join(';') + ';';
       return highlightStyle;
     },
 
     _highlightFilterValue : function(item)
     {
       var highlightStyle = this.__highlightStyle;
-      if (! highlightStyle) {
-        return item;
-      }
 
       var filterValue = this.__filterValue;
-      var parts = item.split(filterValue);
-      if (!this.__lastRich) {
-        parts[0] = qx.module.util.String.escapeHtml(parts[0]);
-        parts[1] = qx.module.util.String.escapeHtml(parts[1]);
+        var parts = item.match(new RegExp('(.*)' +  qx.module.util.String.escapeRegexpChars(filterValue) + '(.*)', 'i'));
+      parts[1] = parts[1] ? parts[1] : '';
+      parts[2] = parts[2] ? parts[2] : '';
+      var label;
+      if (highlightStyle) {
+        filterValue = qx.module.util.String.escapeHtml(this.__filterValue);
```

Comment:
If you asked to have the widget set to 'rich' above, then why to escape the string? Seems sort of the wrong way round. If rich is set, you have to assume that there is html in the labels, so you would want to refrain from escaping and using html of your own. in that case the `|` marker would make sense. I would suggest to use `*` though as this is generally used for emphasis im markdown.

If rich is not set though, you could use it for your own purposes, set it and then use html for highlightning ...

Reply:
- `__filterValue` is user input, thus I think it must be escaped if `rich=true` as it is going to be part of the output to the browser. I guess we could skip it if `rich=false`, but I think it doesn't matter in this case.
- The current implementation assumes that the visible label string itself doesn't contain HTML (it can be inside some HTML tags however). I guess one could try to strip the HTML tags of in both `__filterValue` and the label. I wonder if it is worthwhile (if at all reliable to implement).
I guess we can discuss this in person tomorrow.